### ISSUE TITLE
Add epoch-level callback to a fit() method for monitoring and interoperability

### ISSUE THREAD

**Issue by GiacomoPracucci** [sdv-dev/CTGAN#456]
### Problem Description

The current CTGAN.fit method encapsulates the entire training loop, making it impossible to monitor intermediate metrics (like epoch loss) or intervene during training. This prevents the use of powerful techniques like hyperparameter pruning with tools such as Optuna, advanced logging with MLflow/W&B, or custom early stopping.

### Expected behavior

I propose adding an optional callback parameter to the fit method. This callback would be invoked at the end of each training epoch, receiving the current epoch number and the loss values as arguments.

For example, in the fit method's loop:
```python 
for i in epoch_iterator:
    # ... existing training logic for the epoch ...
    generator_loss = loss_g.detach().cpu().item()
    discriminator_loss = loss_d.detach().cpu().item()
```

```python 
if callback:
    callback(epoch=i, g_loss=generator_loss, d_loss=discriminator_loss)
```

### Additional context

Adding a callback would be a nice quality-of-life improvement for many users, making it much easier to see what's happening during training and connect CTGAN to other tools.

For example, it would allow anyone to:
- Speed up hyperparameter tuning by enabling pruning in libraries like Optuna.
- Log training stats live to services like Weights & Biases or MLflow.
- Implement custom early stopping to avoid wasting time on runs that aren't improving.
- Plot loss curves in real-time to actually watch the model learn.

It's a small change that makes the library far more flexible for anyone building more serious MLOps workflows.

**Comment by GiacomoPracucci** [comment]
I found that this exact functionality is already being tracked in the main SDV repository under a broader initiative: "Support intermediate insights and controls for synthesizer training".

I'm closing this issue.
